Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -73,30 +73,11 @@
     return { { *wgsl, hints } };
 }
 
-static NSString* appleGPUFamilyToString(id<MTLDevice> device)
-{
-#if ENABLE(WEBGPU_BY_DEFAULT)
-    if ([device supportsFamily:MTLGPUFamilyApple9])
-        return @"9";
-    if ([device supportsFamily:MTLGPUFamilyApple8])
-        return @"8";
-#endif
-    if ([device supportsFamily:MTLGPUFamilyApple7])
-        return @"7";
-    if ([device supportsFamily:MTLGPUFamilyApple6])
-        return @"6";
-    if ([device supportsFamily:MTLGPUFamilyApple5])
-        return @"5";
-    if ([device supportsFamily:MTLGPUFamilyApple4])
-        return @"4";
-    return @"0xFF";
-}
-
-id<MTLLibrary> ShaderModule::createLibrary(id<MTLDevice> device, const String& msl, String&& label, NSError** error)
+id<MTLLibrary> ShaderModule::createLibrary(id<MTLDevice> device, const String& msl, String&& label, NSError** error, uint32_t appleGPUFamily)
```

Comment:
Looks like the `appleGPUFamily` parameter here might be redundant? I can only see it being called with `shaderModule.device().appleGPUFamily()`, and further below `device.appleGPUFamily()` is used instead of using the argument.

Reply:
On line 80 there is:
```
    options.preprocessorMacros = @{ @"__wgslMetalAppleGPUFamily" : [NSString stringWithFormat:@"%u", appleGPUFamily] };
```
because we don't have access to the shaderModule inside the static function `id<MTLLibrary> ShaderModule::createLibrary(id<MTLDevice> device, const String& msl, String&& label, NSError** error, uint32_t appleGPUFamily)`

Reply:
Oh, the definition was collapsed and I didn't noticed, so I thought it was in the same function.